pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -349,36 +327,69 @@ func (l *Lifecycle) Updated(obj *v3.KontainerDriver) (runtime.Object, error) {
 		return obj, nil
 	}
 
-	if obj.Spec.BuiltIn {
+	if obj.Spec.BuiltIn && v3.KontainerDriverConditionActive.IsTrue(obj) {
 		// Builtin drivers can still have their schema change during Rancher upgrades so we need to try
 		return obj, l.updateDynamicSchema(obj)
 	}
 
+	var err error
+	var tmpObj runtime.Object
+	newObj := obj.DeepCopy()
```

Comment:
I don't believe that this deep copy is necessary.  You can just return the obj rather than checking for differences at the end of the func, so it's ok to modify the obj we were passed in.